Observer/Customer.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Metrilo\Analytics\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+class Customer implements ObserverInterface
+{
+    /**
+     * @param \Metrilo\Analytics\Helper\Data                    $helper
+     * @param \Metrilo\Analytics\Helper\ApiClient               $apiClient
+     * @param \Metrilo\Analytics\Helper\CustomerSerializer      $customerSerializer
+     * @param \Magento\Customer\Api\CustomerRepositoryInterface $customerRepository
+     */
+    public function __construct(
+        \Metrilo\Analytics\Helper\Data                    $helper,
+        \Metrilo\Analytics\Helper\ApiClient               $apiClient,
+        \Metrilo\Analytics\Helper\CustomerSerializer      $customerSerializer,
+        \Magento\Customer\Api\CustomerRepositoryInterface $customerRepository
+    ) {
+        $this->helper             = $helper;
+        $this->apiClient          = $apiClient;
+        $this->customerSerializer = $customerSerializer;
+        $this->customerRepository = $customerRepository;
+    }
+    
+    public function execute(Observer $observer)
+    {
+        try {
+            $storeId    = $this->helper->getStoreId();
+            $client     = $this->apiClient->getClient($storeId);
```

Comment:
why do you initialize the client so early? What if you don't send the event due to no changes or no email?